src/streams/register_nack.rs

```diff
@@ -619,4 +615,35 @@ mod test {
         let active = reg.active.clone().expect("nack range");
         assert_eq!(*active.start, 65666);
     }
+
+    #[test]
+    fn nack_reports_on_boundaries() {
+        let mut reg = NackRegister::new(None);
+
+        for i in 0..=20 {
+            // gap must be at least 17 packets to separate the reports.
+            if i == 2 || i == 19 {
+                continue;
+            }
+            reg.update(i.into());
+        }
+
+        let reports: Vec<_> = reg
+            .nack_reports()
+            .expect("should generate reports")
+            .flat_map(|nack| nack.reports)
+            .collect();
+
+        assert_eq!(reports.len(), 2, "Should have found two NACK entries");
+        assert_eq!(reports[0].pid, 2, "First missing packet should be 2");
+        assert_eq!(
+            reports[0].blp, 0,
```

Comment:
blp must not be 0 here. If I understand it correctly, it must contain all packages between 2 and 19

Reply:
I think this PR is correct